arrows/vxl/high_pass_filter.cxx

```diff
@@ -0,0 +1,564 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+#include "high_pass_filter.h"
+
+#include <arrows/vxl/image_container.h>
+#include <vital/util/enum_converter.h>
+
+#include <vil/vil_convert.h>
+#include <vil/vil_image_view.h>
+#include <vil/vil_math.h>
+#include <vil/vil_plane.h>
+#include <vil/vil_transpose.h>
+
+namespace kwiver {
+
+namespace arrows {
+
+namespace vxl {
+
+namespace { // anonymous
+
+enum filter_mode
+{
+  MODE_box,
+  MODE_bidir,
+};
+
+ENUM_CONVERTER( mode_converter, filter_mode,
+                { "box", MODE_box }, { "bidir", MODE_bidir } )
+
+// ----------------------------------------------------------------------------
+template <typename T> struct accumulator;
+
+template <> struct accumulator<char>
+{ using type = int; };
```

Comment:
Pedantic, but my understanding is the whitespace between opening and closing <>.

Also, I find particular line is a bit confusing. My understanding is char can be signed or unsigned depending on implementation but int is always signed. So maybe if my understanding is correct, being consistent across this and the following line would be ever so slightly clearer.

Reply:
Derp, and there should be a line break after `template <>`.

I was actually thinking that "signed char" → "signed int" and "char" → "int" *was* consistent :slightly_smiling_face:. I suppose more consistent would be to make `accumulator<char>::type` be signed or unsigned to match `char`, but that complicated to at best gain us an extra bit (since `char` can always promote to `int`).

Reply:
Yeah, no worries with how it is now.